### PR TITLE
Support for Detaching volume in an instance

### PR DESCRIPTION
**Pull request by nvnyale**

### CODE REVIEW

File: api/routes.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func (s *server) routes() {
 	api.HandleFunc("/{account}/volumes/{id}/tags", s.VolumeUpdateHandler).Methods(http.MethodPut)
 
 	api.HandleFunc("/{account}/instances/{id}", s.InstanceDeleteHandler).Methods(http.MethodDelete)
-	api.HandleFunc("/{account}/instances/{id}/volumes/{vid}", s.ProxyRequestHandler).Methods(http.MethodDelete)
+	api.HandleFunc("/{account}/instances/{id}/volumes/{vid}", s.VolumeDetachHandler).Methods(http.MethodDelete).Queries("force", "{force}")
```
Comment (tenyo) [inline api/routes.go:80]: I think using `Queries` here makes `force` a required parameter, but I'm not sure if we always pass it. You may want to test from the UI

File: api/handlers_instances.go [hunk 2]
```diff
@@ -550,3 +550,42 @@ func (s *server) InstanceUpdateHandler(w http.ResponseWriter, r *http.Request) {
...
+		handleError(w, err)
+		return
+	}
+
+	out, err := orch.detachVolume(r.Context(), instance_id, volume_id, force)
+	if err != nil {
+		handleError(w, err)
+		return
+	}
+		handleResponseOk(w, out)
```
[the remark this answers is not in the document]
Comment (nvnyale) [reply api/handlers_instances.go]: done.

File: ec2/instances_test.go [hunk 3]
```diff
@@ -503,3 +513,56 @@ func TestEc2_RebootInstance(t *testing.T) {
 		})
 	}
 }
+
+func TestEc2_DetachVolume(t *testing.T) {
+	type fields struct {
+		//session *session.Session
```
[the remark this answers is not in the document]
Comment (nvnyale) [reply ec2/instances_test.go]: done.

File: api/routes.go (hunk 1 above)
Comment (nvnyale) [reply api/routes.go]: I have made the `FORCE` query parameter optional in the code. Please review.

**Review by tenyo — APPROVED** [review]
Looks good 👍

**Pull request merged**